### PR TITLE
Port FSTListenOptions to C++

### PR DESCRIPTION
**Pull request by wilhuff**
Note the DIFFBASE on #2608.

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/core/listen_options.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 Google
```
Comment (mikelehen) [inline Firestore/core/src/firebase/firestore/core/listen_options.h]: 2019?

File: Firestore/core/src/firebase/firestore/core/listen_options.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+   */
+  ListenOptions(bool include_query_metadata_changes,
+                bool include_document_metadata_changes,
+                bool wait_for_sync_when_online)
+      : include_query_metadata_changes_(include_query_metadata_changes),
+        include_document_metadata_changes_(include_document_metadata_changes),
+        wait_for_sync_when_online_(wait_for_sync_when_online) {
+  }
+
+  /** Converts the public API options object to the internal options object. */
```
Comment (mikelehen) [inline Firestore/core/src/firebase/firestore/core/listen_options.h]: stale comment?

File: Firestore/core/src/firebase/firestore/core/listen_options.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  static ListenOptions DefaultOptions() {
+    return ListenOptions(
+        /*include_query_metadata_changes=*/false,
+        /*include_document_metadata_changes=*/false,
+        /*wait_for_sync_when_online=*/false);
+  }
+
+  /** Converts the public API options object to the internal options object. */
```
Comment (mikelehen) [inline Firestore/core/src/firebase/firestore/core/listen_options.h]: stale comment?

File: Firestore/core/src/firebase/firestore/core/listen_options.h (hunk 1 above)
Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/core/listen_options.h]: Done.

File: Firestore/core/src/firebase/firestore/core/listen_options.h (hunk 2 above)
Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/core/listen_options.h]: Done.

File: Firestore/core/src/firebase/firestore/core/listen_options.h (hunk 3 above)
Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/core/listen_options.h]: Done.

**Pull request merged**